system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts

```diff
@@ -0,0 +1,406 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {
+  IHederaTokenServiceKeyType,
+  TokenQuerySmartContractResult,
+} from '@/types/contract-interactions/HTS';
+import { handleContractResponseWithDynamicEventNames } from '@/utils/contract-interactions/HTS/helpers';
+import { KEY_TYPE_MAP } from '@/utils/contract-interactions/HTS/token-create-custom/constant';
+import { Contract, isAddress } from 'ethers';
+
+/**
+ * @dev queries token validity
+ *
+ * @dev integrates TokenQueryContract.isTokenPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @return Promise<TokenQuerySmartContractResult>
+ */
+export const queryTokenValidity = async (
+  baseContract: Contract,
+  hederaTokenAddress: string
+): Promise<TokenQuerySmartContractResult> => {
+  // sanitize param
+  if (!isAddress(hederaTokenAddress)) {
+    console.error('Invalid token address');
+    return { err: 'Invalid token address' };
+  }
+
+  try {
+    const transactionResult = await baseContract.isTokenPublic(hederaTokenAddress);
+    // get transaction receipt
+    const txReceipt = await transactionResult.wait();
+
+    // retrieve information from event
+    const { data } = txReceipt.logs.filter((event: any) => event.fragment.name === 'IsToken')[0];
+    return { IsToken: data, transactionHash: txReceipt.hash };
+  } catch (err: any) {
+    console.error(err);
+    return { err, transactionHash: err.receipt && err.receipt.hash };
+  }
+};
+
+/**
+ * @dev queries token general information
+ *
+ * @dev integrates TokenQueryContract.getTokenInfoPublic()
+ *
+ * @dev integrates TokenQueryContract.getFungibleTokenInfoPublic()
+ *
+ * @dev integrates TokenQueryContract.getNonFungibleTokenInfoPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param API: "TOKEN_INFO" | "FUNGIBLE_INFO" | "NON_FUNFIBLE_INFO"
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param serialNumber?: number
+ *
+ * @return Promise<TokenQuerySmartContractResult>
+ */
+export const queryTokenGeneralInfomation = async (
+  baseContract: Contract,
+  API: 'TOKEN_INFO' | 'FUNGIBLE_INFO' | 'NON_FUNFIBLE_INFO',
+  hederaTokenAddress: string,
+  serialNumber?: number
+): Promise<TokenQuerySmartContractResult> => {
+  // sanitize param
+  if (!isAddress(hederaTokenAddress)) {
+    console.error('Invalid token address');
+    return { err: 'Invalid token address' };
+  } else if (serialNumber && serialNumber < 0) {
+    console.error('Invalid serial number');
+    return { err: 'Invalid serial number' };
+  }
+
+  // prepare events map
+  const eventMaps = {
+    TOKEN_INFO: 'TokenInfo',
+    FUNGIBLE_INFO: 'FungibleTokenInfo',
+    NON_FUNFIBLE_INFO: 'NonFungibleTokenInfo',
+  };
+
+  // invoking contract methods
+  try {
+    let transactionResult;
+    switch (API) {
```

Comment:
All switches should have a default to catch unexpected cases.

Reply:
I specifically set the `API` parameter to include only those values, which is why I didn't add a `default` case at the end for for the switches.